EIPS/eip-2566.md

```diff
@@ -0,0 +1,131 @@
+          "name": "_address",
+          "type": "address"
+        }, {
+          "name": "_value",
+          "type": "uint256"
+        }],
+        "name": "transferTokens",
+        "outputs": [{
+          "name": "success",
+          "type": "bool"
+        }],
+        "stateMutability": "nonpayable",
+        "type": "function"
+      }",
+      "args": [
+        "0x6Aa89e52c9a826496A8f311c1a9db62fd477E256",
+        100000000000
+      ]
+    }
+}]
+```
+
+Returns
+DATA, 32 Bytes - the transaction hash, or the zero hash if the transaction is not yet available.
+
+Example
+// Request
+curl -X POST --data '{"jsonrpc":"2.0","method":"eth_sendTransactionToContractFunction","params":[{see above}],"id":1}'
+
+// Result
+{
+  "id":1,
+  "jsonrpc": "2.0",
+  "result": "0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331"
+}
+
+## Rationale
+With backwards compatibility in mind, this EIP proposes augmenting the set of Ethereum RPC methods with an additional method instead of mutating the existing paralleled method. Precedent for adding a new RPC method comes from [EIP 712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md) in which adding the method `eth_signTypedData` is proposed for confirmation prompt security. As an alternate approach, the `eth_sendTransaction` method could be changed to accept its `data` argument in a human readable format, but this would break all existing code attempting to execute a contract function. Breaking this backwards compatibility would be a far bigger issue than the readability of an `data` argument so this EIP does not recommended this approach.
+
+This EIP's proposed `eth_sendTransactionToContractFunction` method is intended to parallel `eth_sendTransaction` as much as possible since both methods result in the same behaviour when executing a contract function. This slightly vague `data` field was renamed to `function` to improve specificity. The `function` field accepts an object of its own with two required fields: `abi` and `args`. The `abi` field is an element of the contract's ABI that corresponds to the intended function. The `args` field is a non ABI-encoded array of arguments to be passed to the contract function. Dapps need not ABI-encode arguments as the data types of these arguments can be inferred from the `abi` field on the ProviderWallet side and encoded at that time.  
+
+## Backwards Compatibility
+This EIP does not break backwards compatibility and addresses that point above.
+
+## Implementation
+The `data` field from `eth_sendTransaction` can be reconstructed in its non-human readable format from the `abi` and `args` fields in `eth_sendTransactionToContractFunction`. This allows implementations to broadcast a transaction with data in the correct format for the Ethereum virtual machine. The function signature can be built and hashed from the function ABI, and the arguments can be ABI-encoded according to the data types specified in the function ABI.
```

Comment:
As state above, what is the benefit of having an new way of encoding the args? Given the abi, `eth_sendTransactionToContractFunction` should be able to parse `args` just fine.

Reply:
Yup you're correct - will make the change.